sprocs/workspace_loads_current.sql

```diff
@@ -98,8 +106,15 @@ BEGIN
     FROM workspace_hosts AS wh
     WHERE wh.state = 'terminating';
 
-    -- Compute desired number of workspace hosts
-    workspace_jobs_capacity_desired := workspace_active_count * workspace_capacity_factor;
+    -- Compute desired number of workspace hosts.
+    --
+    -- We only consider workspaces that are launching OR that are running on
+    -- a healthy host. In the case where a workspace is running on a host that
+    -- is unhealthy, that workspace will continue to run there until it times
+    -- out and is shut down. We don't want to include those workspaces in the
+    -- computation to avoid overprovisioning after deploys when all existing
+    -- hosts are marked as unhealthy.
+    workspace_jobs_capacity_desired := (workspace_running_on_healthy_hosts_count + workspace_launching_count) * workspace_capacity_factor;
```

Comment:
Do we want to compute
```
workspace_active_on_healthy_host_count := workspace_running_on_healthy_hosts_count + workspace_launching_count
```
and output this to CloudWatch? That would allow us to better visualize the metric that is actually being used for autoscaling.

Reply:
Sure, that sounds reasonable!

Reply:
Done in 64288c2.